posts/google-summer-of-code-fellows/gsoc-2024-fellows.qmd

```diff
@@ -0,0 +1,294 @@
+
+4. Create Continuous integration and Continuous Development pipeline for the package. 
+
+5. Implement the combinationWindow function, that combines treatments with various overlapping strategies.
+
+// Stretch Goals:
+
+(a) Composing with JuliaStats Ecosystem
+(b) Novel Visualizations for Pathways
+
+
+# Work Details
+
+**1. Setting up the package in JuliaHealth Channel**
+
+Initially, there was no package as such for pathways synthesis, so I had to build it from scratch. Firstly, the repository by the name [OMOPCDMPathways.jl](https://github.com/JuliaHealth/OMOPCDMPathways.jl). Once the repository was created, we needed to have a skeleton for a standard Julia repository. For this, we used the [PkgTemplate.jl](https://juliaci.github.io/PkgTemplates.jl/stable/user/) this provided a basic skeleton for the repository that included - folders for test suites, documentation, src [code files], GitHub files, Readme and Licensing file, toml and citation files. All this we can further edit and modify as per our work. By default, PkgTemplate.jl uses [Documenter.jl](https://documenter.juliadocs.org/stable/) for the documentation part but as suggested and discussed with my mentor we decided to shift to [DocumenterVitepress.jl](https://luxdl.github.io/DocumenterVitepress.jl/dev/) for the documentation part. However, we still faced some deployment issues in the new documentation thanks to Anshul for helping fix the [Deployment issues with DocumenterVitepress](https://github.com/JuliaHealth/OMOPCDMPathways.jl/issues/15). With this, we were ready with the documentation set up and fully functional. After we had shifted to DocumenterVitepress the main task now was to host the documentation, this was done using Github-Actions, detailed steps for hosting are provided at [this](https://documenter.juliadocs.org/stable/man/hosting/#Hosting-Documentation) page. Then we added the codecov to our package by triggering it via a dummy function and a corresponding test case for it. Also, the CI for the package was set up with it. And, now finally the repository was ready with test coverage, CI, and documentation fully functional repository ready.
+Initial documentations: ![](./image.png)
+New documentations using DocumenterVitepress: ![](./img2.png)
+
+
+
+**2. Loading the PostgreSQL Database**
+
+The main database we worked on/built analysis was the freely available OMOPCDM Database. However, I was provided with some more extra synthetic data from my mentor for further testing of the functionalities. Being a very large database we had to strategically download it further, my mentor helped me in setting up the Postgres on my local machine. Once, the database was set up proper testing was performed on it to check if things were as expected. With this, we were done with the database setup as well and could finally dive into the actual code logic for the Pathways synthesis.
```

Comment:
We can actually put your report on Zenodo in a little bit so as to link the database too which will be neat.